eclass/user-info.eclass

```diff
@@ -37,18 +37,27 @@ egetent() {
 		# lookup by uid/gid
 		local opts
 		if [[ ${key} == [[:digit:]]* ]] ; then
-			[[ ${db} == "user" ]] && opts="-u" || opts="-g"
+			[[ ${db} == "user" ]] && opts=( "-u" ) || opts=( "-g" )
 		fi
 
+		# Handle different ROOT
+		[[ ! -z ${ROOT} ]] && opts+=( -R "${ROOT}" )
+
 		pw show ${db} ${opts} "${key}" -q
 		;;
 	*-openbsd*)
-		grep "${key}:\*:" /etc/${db}
+		grep "${key}:\*:" "${ROOT}/etc/${db}"
 		;;
 	*)
-		# ignore nscd output if we're not running as root
-		type -p nscd >/dev/null && nscd -i "${db}" 2>/dev/null
-		getent "${db}" "${key}"
+		# getent does not support -R option, if we are working on a different
+		# ROOT than /, fallback to grep technique.
+		if [[ -z ${ROOT} ]]; then
+			# ignore nscd output if we're not running as root
+			type -p nscd >/dev/null && nscd -i "${db}" 2>/dev/null
+			getent "${db}" "${key}"
+		else
+			grep "^${key}" "${ROOT}/etc/${db}"
```

Comment:
This regex is definitely too weak — `egetent passwd foo` would match `foobar`.

Reply:
I guess that: "^${key}:.*:" should do the trick.
+ I need to replace ROOT by EROOT like in the previous comment ?

Reply:
Yes, I guess so. Maybe even `^${key}:`